src/data-operations/filtering-condition.ts

```diff
@@ -0,0 +1,278 @@
+// helper functions
+function applyIgnoreCase (a: string, ignoreCase: boolean) : string {
+    a = a || "";
+    // bulletproof
+    return ignoreCase? ("" + a).toLowerCase() : a;
+}
+function getDateParts(date: Date, dateFormat?: string): 
+                    {   year?: number, 
+                        month?: number, 
+                        day?: number, 
+                        hours?: number, 
+                        minutes?: number, 
+                        seconds?: number, 
+                        milliseconds?: number
+                    } {
+    let res = {
+        year: null, 
+        month: null, 
+        day: null, 
+        hours: null, 
+        minutes: null, 
+        seconds: null, 
+        milliseconds: null
+    };
+    if (!date || !dateFormat) {
+        return res;
+    }
+    if (dateFormat.indexOf("y") >= 0) {
+        res.year = date.getFullYear();
+    }
+    if (dateFormat.indexOf("M") >= 0) {
+        res.month = date.getMonth();
+    }
+    if (dateFormat.indexOf("d") >= 0) {
+        res.day = date.getDate();
+    }
+    if (dateFormat.indexOf("h") >= 0) {
+        res.hours = date.getHours();
+    }
+    if (dateFormat.indexOf("m") >= 0) {
+        res.minutes = date.getMinutes();
+    }
+    if (dateFormat.indexOf("s") >= 0) {
+        res.seconds = date.getSeconds();
+    }
+    if (dateFormat.indexOf("f") >= 0) {
+        res.milliseconds = date.getMilliseconds();
+    }
+    return res;
+}
+
+export declare type FilteringConditionArgs = {
```

Comment:
This should go away. Leave the value and the ignoreCase in the string case, and only the value for the other types. This object will be hard to populate by a developer.